packages/participants-svc/src/domain/participant_agg.ts

```diff
@@ -917,34 +916,182 @@ export class ParticipantAggregate {
+        );
+
+        await this._auditClient.audit(
+            AuditedActionNames.PARTICIPANT_ACCOUNT_CHANGE_REQUEST_CREATED,
+            true,
+            this._getAuditSecCtx(secCtx),
+            [{ key: "participantId", value: participantId }]
+        );
+
+        return accountChangeRequest.id;
+    }
+
+    async approveParticipantAccount(
+        secCtx: CallSecurityContext,
+        participantId: string,
+        accountChangeRequestId: string
+    ): Promise<string> {
+        this._enforcePrivilege(secCtx, ParticipantPrivilegeNames.APPROVE_ACCOUNTS_CHANGE_REQUEST);
 
-            throw new UnableToCreateAccountUpstream(
-                `'${existing.name}' account '${account.type}' failed upstream.`
+        if (!participantId)
+            throw new InvalidParticipantError("[id] cannot be empty");
+
+        const existing: IParticipant | null = await this._repo.fetchWhereId(
+            participantId
+        );
+        if (!existing)
+            throw new ParticipantNotFoundError(
+                `Participant with ID: '${participantId}' not found.`
+            );
+        // if (!existing.isActive) throw new ParticipantNotActive("Participant is not active.");
+
+        const accountChangeRequest = existing.participantAccountsChangeRequest.find(
+            (value: IParticipantAccountsChangeRequest) => value.id === accountChangeRequestId
+        );
+        if (!accountChangeRequest) {
+            throw new AccountNotFoundError(
+                `Cannot find a participant's account change request with id: ${accountChangeRequestId}`
+            );
+        }
+        if (accountChangeRequest.approved) {
+            throw new AccountNotFoundError(
+                `Participant's account change request with id: ${accountChangeRequestId} is already approved`
             );
         }
 
-        existing.participantAccounts.push({
-            id: createdId,
-            type: account.type as ParticipantAccountTypes,
-            currencyCode: account.currencyCode,
-            creditBalance: null,
-            debitBalance: null,
-            balance: null
-        });
+        // now we can enforce the correct privilege
+        this._enforcePrivilege(
+            secCtx,
+            ParticipantPrivilegeNames.APPROVE_ACCOUNTS_CHANGE_REQUEST
+        );
+
+        if (secCtx && accountChangeRequest.createdBy === secCtx.username) {
+            await this._auditClient.audit(
+                ParticipantChangeTypes.CHANGE_ACCOUNT,
+                false,
+                this._getAuditSecCtx(secCtx),
+                [{ key: "participantId", value: participantId }]
+            );
+            throw new MakerCheckerViolationError(
+                "Maker check violation - Same user cannot create and approve participant account change request"
+            );
+        }
+
+        if (!existing.participantAccounts) {
+            existing.participantAccounts = [];
+        }
+        // else {
```

Comment:
When the change request is for creation/add we should check that a similar account doesn't exist already

Reply:
Added validation with existing account type when incoming new account creation.